muikku-core-plugins/src/main/resources/META-INF/resources/jsf/announcements/index.xhtml

```diff
@@ -1,17 +1,23 @@
-<ui:composition xmlns="http://www.w3.org/1999/xhtml"
+<ui:composition
+  xmlns="http://www.w3.org/1999/xhtml"
   xmlns:ui="http://java.sun.com/jsf/facelets"
   xmlns:h="http://java.sun.com/jsf/html"
   xmlns:jsf="http://xmlns.jcp.org/jsf"
-  template="/WEB-INF/templates/base.xhtml">
-  
+  template="/WEB-INF/templates/base.xhtml"
+>
   <ui:define name="scripts">
-    <script defer="defer" src="#{request.contextPath}/scripts/dist/vendor.js" />
-    <script defer="defer" src="#{request.contextPath}/scripts/dist/main-function.js" />
+    <script
+      defer="defer"
+      src="#{request.contextPath}/scripts/dist/main-function.js"
+      type="module"
```

Comment:
Jos käytetään module arvoa type attribuutissa niin defer ja charset attribuuteilla ei ole vaikutusta. Samaten CORS pitää ottaa huomioon.

_Type value causes the code to be treated as a JavaScript module. The processing of the script contents is deferred. The charset and defer attributes have no effect. For information on using module, see our [JavaScript modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) guide. Unlike classic scripts, module scripts require the use of the CORS protocol for cross-origin fetching._

[https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type](url)